pandas/tests/frame/test_constructors.py

```diff
@@ -314,6 +314,17 @@ def test_constructor_dict(self):
         with pytest.raises(ValueError, match=msg):
             DataFrame({'a': 0.7}, columns=['a'])
 
+    def test_constructor_dict_order(self):
+        data = {'B': {'b': 1, 'c': 2, 'a': 3},
+                'A': {'b': 3, 'c': 2, 'a': 1}}
```

Comment:
What happens if the order isn't the same across all dicts?

Reply:
It would keep the order of the first dict. For example if the first dict is ordered b, c, a like in the above example and the second dict is ordered c, b, a - the resulting DataFrame would maintain the order b, c, a.